examples/vision/attention_mil_classification.py

```diff
@@ -0,0 +1,589 @@
+matrix, **u**, is added to the computation.
+Sigmoid non-linearity is used to overcome approximately linear behavior for *x* ∈ [−1, 1]
+by hyperbolic tangent non-linearity.
+"""
+
+
+class MILAttentionLayer(layers.Layer):
+    """Implementation of the attention-based Deep MIL layer.
+
+    Args:
+      weight_params_dim: Positive Integer. Dimension of the weight matrix.
+      kernel_initializer: Initializer for the `kernel` matrix.
+      kernel_regularizer: Regularizer function applied to the `kernel` matrix.
+      use_gated: Boolean, whether or not to use the gated mechanism.
+
+    Returns:
+      List of 2D tensors with BAG_SIZE length.
+      The tensors are the attention scores after softmax with shape `(batch_size, 1)`.
+    """
+
+    def __init__(
+        self,
+        weight_params_dim,
+        kernel_initializer="glorot_uniform",
+        kernel_regularizer=None,
+        use_gated=False,
+        **kwargs,
+    ):
+
+        super().__init__(**kwargs)
+
+        self.weight_params_dim = weight_params_dim
+        self.use_gated = use_gated
+
+        self.kernel_initializer = keras.initializers.get(kernel_initializer)
+        self.kernel_regularizer = keras.regularizers.get(kernel_regularizer)
+
+        self.v_init = self.kernel_initializer
+        self.w_init = self.kernel_initializer
+        self.u_init = self.kernel_initializer
+
+        self.v_regularizer = self.kernel_regularizer
+        self.w_regularizer = self.kernel_regularizer
+        self.u_regularizer = self.kernel_regularizer
+
+    def build(self, input_shape):
+
+        # Input Shape
+        # List of 2D tensors with shape: (batch_size, input_dim).
+        input_dim = input_shape[0][1]
+
+        self.v_weight_params = self.add_weight(
+            shape=(input_dim, self.weight_params_dim),
+            initializer=self.v_init,
+            name="v",
+            regularizer=self.v_regularizer,
+            trainable=True,
+        )
+
+        self.w_weight_params = self.add_weight(
+            shape=(self.weight_params_dim, 1),
+            initializer=self.w_init,
+            name="w",
+            regularizer=self.w_regularizer,
+            trainable=True,
+        )
+
+        if self.use_gated:
+            self.u_weight_params = self.add_weight(
+                shape=(input_dim, self.weight_params_dim),
+                initializer=self.u_init,
+                name="u",
+                regularizer=self.u_regularizer,
+                trainable=True,
+            )
+        else:
+            self.u_weight_params = None
+
+        self.input_built = True
+
+    def call(self, inputs):
+
+        # Assigning variables from the number of inputs.
+        instances = [self.compute_weights(instance) for instance in inputs]
+
+        # such that each row summation is equal to 1.
+        alpha = tf.math.softmax(instances, axis=0)
+
+        return [alpha[i] for i in range(alpha.shape[0])]
+
+    def compute_weights(self, instance):
```

Comment:
maybe rename this `compute_attention_scores`? the weights of this layer are not actually what you are computing, that won't happen until gradients are applied.

Reply:
Updated it accordingly.